src/main/resources/+ciplugins/+jenkins/BuildReportPlugin.m

```diff
@@ -6,20 +6,24 @@
 
         function runTaskGraph(plugin, pluginData)
             runTaskGraph@matlab.buildtool.plugins.BuildRunnerPlugin(plugin, pluginData);
-            fID = fopen(fullfile(getenv("WORKSPACE"),'.matlab/buildArtifact.json'), 'w');
-            taskDetails = struct();
-            for idx = 1:numel(pluginData.TaskResults)
-                taskDetails(idx).name = pluginData.TaskResults(idx).Name;
-                taskDetails(idx).description = pluginData.TaskGraph.Tasks(idx).Description;
-                taskDetails(idx).failed = pluginData.TaskResults(idx).Failed;
-                taskDetails(idx).skipped = pluginData.TaskResults(idx).Skipped;
-                taskDetails(idx).duration = string(pluginData.TaskResults(idx).Duration);
+            [fID, msg] = fopen(fullfile(getenv("WORKSPACE"),'.matlab/buildArtifact.json'), 'w');
+
+            if fID == -1
+                warning('BuildTool:artifactFileWarnning','Could not open a file for Jenkins build result table due to: %s', msg);
+            else
```

Comment:
Agree, And thats why prolly we could not catch it. I will continue the review this more to figure out what was the root cause before having a fullproof fix. So far I could not reproduce it on local. this fix is just to unblock current users from failing the build. Jenkins shows following with this fix. after due to : it will print actual fopen error.

![image](https://github.com/mathworks/jenkins-matlab-plugin/assets/47204011/d7abfa05-fe07-4e06-99eb-7f8c3204a59b)